pkg/reconciler/serverlessservice/serverlessservice.go

```diff
@@ -254,15 +254,16 @@ func (r *reconciler) reconcilePublicEndpoints(ctx context.Context, sks *netv1alp
 	} else if !metav1.IsControlledBy(eps, sks) {
 		sks.Status.MarkEndpointsNotOwned("Endpoints", sn)
 		return fmt.Errorf("SKS: %s does not own Endpoints: %s", sks.Name, sn)
-	}
-	want := eps.DeepCopy()
-	want.Subsets = srcEps.Subsets
-
-	if !equality.Semantic.DeepEqual(want.Subsets, eps.Subsets) {
-		logger.Info("Public K8s Endpoints changed; reconciling: ", sn)
-		if _, err = r.KubeClientSet.CoreV1().Endpoints(sks.Namespace).Update(want); err != nil {
-			logger.Errorw("Error updating public K8s Endpoints: "+sn, zap.Error(err))
-			return err
+	} else {
+		wantSubsets := resources.FilterSubsetPorts(sks, srcEps.Subsets)
+		if !equality.Semantic.DeepEqual(wantSubsets, eps.Subsets) {
```

Comment:
cc @tcnghia 

Will this impact the e2e tests that look for matching endpoints?

Reply:
This might. We'll see once the integration test run finishes.

Reply:
I might need to change that to compare addresses only

Reply:
Fixed and checked.